api/data.go

```diff
@@ -180,3 +200,24 @@ func isHexadecimal(s string) bool {
 func isLower(s string) bool {
 	return strings.ToLower(s) == s
 }
+
+func sendJSONError(ctx context.Context, w http.ResponseWriter, code int, message string) {
+	w.WriteHeader(code)
+	if err := json.NewEncoder(w).Encode(map[string]string{"error": message}); err != nil {
+		log.Error(ctx, "error encoding results to JSON", err)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		return
+	}
+}
+
+func formatErrorListToArrayString(errList []error) string {
```

Comment:
This isn't returning an array of strings, so it should probably be renamed to `formatErrorList` or similar.